applications/RANSApplication/custom_elements/convection_diffusion_reaction_cross_wind_stabilized_element.h

```diff
@@ -0,0 +1,989 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Suneth Warnakulasuriya
+//
+
+#if !defined(KRATOS_CONVECTION_DIFFUSION_REACTION_CROSS_WIND_STABILIZED_ELEMENT_H_INCLUDED)
+#define KRATOS_CONVECTION_DIFFUSION_REACTION_CROSS_WIND_STABILIZED_ELEMENT_H_INCLUDED
+
+// System includes
+#include <cmath>
+
+// External includes
+
+// Project includes
+#include "includes/cfd_variables.h"
+#include "includes/checks.h"
+#include "includes/define.h"
+#include "includes/element.h"
+#include "utilities/geometry_utilities.h"
+#include "utilities/time_discretization.h"
+
+// Application includes
+#include "convection_diffusion_reaction_stabilization_utilities.h"
+#include "custom_utilities/rans_calculation_utilities.h"
+
+namespace Kratos
+{
+///@name Kratos Classes
+///@{
+
+template <unsigned int TDim,
+          unsigned int TNumNodes,
+          class TConvectionDiffusionReactionData>
+class ConvectionDiffusionReactionCrossWindStabilizedElement
+: public Element
```

Comment:
Shouldn't this derive from the `ConvectionDiffusionReactionElement`? At first glance some of the method seem equivalent to me.

Reply:
That is doable. I already have some kind of little bit not so easy to understand templating inside (this will get worse when I push my adjoint PR for all these elements, thats waiting till this is gone through). I also like the idea of reducing code duplication, I try always to reduce number of lines. At somepoint it becomes little bit harder to read. I can change it with some additional functions for sure. What's your opinion on this? :)

Reply:
I'd be in favor of reducing code duplication as much as possible. I think it should be possible if we introduce some auxiliary methods to be override.

Reply:
done :)